---
title: "x"
author: "Parth"
date: "2024-05-31"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
install.packages("tidyverse")
library(tidyverse)
library(ggplot2)
library(dplyr)
```
```{r}
if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

BiocManager::install("DESeq2")
```


```{r}
install.packages("BiocManager")
BiocManager::install("DESeq2")
library(DESeq2)
```




```{r}
library(readr)
library(readr)
 setwd("C:/Users/parth/OneDrive/Desktop/repet_1")
# Read CSV file with the first column as row names
DatasheetB <- read.csv("C:/Users/parth/OneDrive/Desktop/New_start/DatasheetB.csv", header = TRUE, row.names = 1, sep = ",")

# View the data
```


```{r}


DatasheetB



```


```{r}
if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
```


```{r}
BiocManager::install("pasilla")
```
Exclude text columns before you proceed save them seperately

```{r}
condition<-factor(c("D","D","M","M","D"))
```
Mention the conditon and equate column anems to row names 
Sort the data in metastatic and dormant models

```{r}
coldata<-data.frame(row.names=colnames(DatasheetB),condition)
```

```{r}
coldata
```

```{r}
dds<-DESeqDataSetFromMatrix(countData = round(DatasheetB),colData = coldata,design = ~condition)
```


```{r}
dds<-DESeq(dds)
```
pull out individual values
```{r}
dds$condition
dds$sizeFactor

```

```{r}
colData(dds)
design(dds)<-~condition
```

Now do the real analysis 
```{r}
de_dormancy<-DESeq(dds)
```
find the results 
```{r}
results(de_dormancy)
```
Tidy up the data now abd start using p value for getting significant difference and write a file about it to get everything sorted formats: excel or CSV (here csv)
```{r}
library(dplyr)
library(tidyverse)
library(DESeq2)
dir.create("BCanalysis",showWarnings = FALSE)
results(de_dormancy,tidy = TRUE)%>%
  arrange(desc(padj>0.05))%>%
  write_csv("dormancy_analysis.csv")
read_csv("dormancy_analysis.csv")
saveRDS(de_dormancy,file = "de_dormancy.rds")
```


Finally plot the data for the first time
```{r}
# Load the DESeq2 library
library(DESeq2)

# Assuming de_dormancy is your DESeqDataSet object and you've already run DESeq
# de_dormancy <- DESeq(de_dormancy)

# Get the results
res <- results(de_dormancy)

# Filter for adjusted p-values less than 0.05, and ensure there are no NA values
filtered_res <- res[!is.na(res$padj) & res$padj < 0.05, ]

# Check if the filtered results are not empty
if (nrow(filtered_res) > 0) {
    # Plot the MA plot using the filtered results
    plotMA(filtered_res, main="MA plot with adjusted p-value < 0.05")
    
    # Optional: Highlight significant points
    plotMA(res, ylim=c(-2, 2))
    with(subset(res, !is.na(padj) & padj < 0.05), points(baseMean, log2FoldChange, col="red", cex=1.5))
} else {
    cat("No genes passed the adjusted p-value threshold of 0.05\n")
}





```
```{r}
plotMA(de_dormancy)
```




Now annotations to reveal gene expressions, this annotates gene expression to human genome  
```{r}
BiocManager::install("org.Hs.eg.db")
# For Human

```

Now checkout the columns 
```{r}
library(org.Hs.eg.db)
columns(org.Hs.eg.db)

```
```{r}
keys(org.Hs.eg.db, keytype="ENSEMBL")[1:10]
```
Analysing gene function and alloting them gene name and symbols
```{r}
AnnotationDbi::select(org.Hs.eg.db, keys="ENSG00000000003", keytype="ENSEMBL", columns=c("SYMBOL", "GENENAME"))

```
Now we annotate the whole bunch of data 
```{r}
anno<-AnnotationDbi::select(org.Hs.eg.db,keys = rownames(dds),columns = c("SYMBOL","GENENAME"),keytype = "ENSEMBL")
View(anno)
dim(anno)
dim(dds)
```
The dimmensions of anno and dds dont match due to the duplications in the dataset 
Now to remove duplicated information 

```{r}
a<-duplicated(dds)
sum(a)
```
Now to remove the duplicated entries 
```{r}
anno<-AnnotationDbi::select(org.Hs.eg.db,keys = rownames(dds),columns = c("ENSEMBL","SYMBOL","GENENAME","ENTREZID"),keytype="ENSEMBL")%>%filter(!duplicated(ENSEMBL))
head(anno)
```
Now joining the dataframes together with gene names and database. This joins the ENSEMBL datasheet with gene names and gene symbols 
```{r}
results_annotated<-results(de_dormancy,tidy = TRUE)%>%left_join(anno,by=c("row"="ENSEMBL"))
head(results_annotated)
write.csv(results_annotated,file = "de_dormancy-MvsD.csv",row.names = FALSE)
saveRDS(results_annotated,file = "dormancy_MvsD.rds")
```
Normalising the datasets
```{r}
dds<-estimateSizeFactors(dds)
countMatrix <-counts(dds, normalized=TRUE)
View(countMatrix)
write.csv(countMatrix,file= "normalised counts.csv")

```

To the final stage of heat mapp analysis and data analysis 

```{r}
library(DESeq2)
results_DM<-readRDS("C:/Users/parth/OneDrive/Desktop/New_start/dormancy_MvsD.rds")
head(results_DM)
summary(results_DM)
# Remove rows with any NA values
clean_data <- results_DM[complete.cases(results_DM), ]

```
Just create ggplots 
```{r}
clean_data %>% 
  mutate(Significant = padj < 0.05 & abs(log2FoldChange) > 2) %>% 
  ggplot(aes(x = log2FoldChange, y = padj, col=Significant)) + geom_point()
```
Create some volcano plots and see if you find some data
```{r}
N<-10
View(clean_data)
top_genes <- dplyr::slice(clean_data, 1:N) %>% pull(row)

clean_data %>%
  mutate(label = ifelse(row %in% top_genes, SYMBOL, "")) %>%
  ggplot(aes(x = log2FoldChange, y = padj, label = label)) +
  geom_point(alpha = 0.4) +
  geom_text(col = "blue")

if(!require(ggrepel)) install.packages("ggrepel")

clean_data %>% 
  mutate(Label = ifelse(row %in% top_genes, SYMBOL, "")) %>%  
  ggplot(aes(x = log2FoldChange, y = padj, label=Label)) + geom_point(alpha=0.4) + geom_text_repel(col="blue")
  
  
```

now creating heatmaps
```{r}
install.packages("pheatmap")
library(pheatmap)
top_genes <- dplyr::slice(clean_data, 1:N) %>% pull(row)
vsd<-vst(dds)
pheatmap(assay(vsd)[top_genes,])

```
For some reason the annotation is not working and its giving me an error about incorrect dimensions 
Annotation is working apparently ignore annotation column 
```{r}
sampleInfo <- colData(dds)[, c("condition")]
print(sampleInfo)

gene_labels <- dplyr::slice(results_DM, 1:N) %>% pull(SYMBOL)
pheatmap(assay(vsd)[top_genes,],
           labels_row = gene_labels,
         
         scale="row")
```



Now do pathway analysis 
```{r}
library(org.Hs.eg.db)
pathway_genes <- AnnotationDbi::select(org.Hs.eg.db,
                                       keys = "GO:0030198",
                                       keytype = "GO",
                                       columns="ENSEMBL") %>% pull(ENSEMBL)

```


```{r}
library(dplyr)
library(org.Hs.eg.db)
go_table <- mutate(results_DM, 
                   inPathway = row %in% pathway_genes,
                   isDE = padj < 0.05 & abs(log2FoldChange) > 1)
View(go_table)
```

```{r}
table(go_table$inPathway, go_table$isDE)
```
Performing Chisquare test on the data 
```{r}
chisq.test(table(go_table$inPathway, go_table$isDE))
```
The test resulted in very small negative p value hence the alternative hypothesis stating there is high change in gene expression in dormant and metastatic cells is seen 

Clusterprofiler analysis 
```{r}
if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

BiocManager::install("clusterProfiler")



```


```{r}
print(head(results_DM$row
           ))
```


```{r}
library(clusterProfiler)
universe <- results_DM %>% pull(row)
sigGenes <- results_DM %>% 
  filter(padj < 0.05, !is.na(row)) %>% pull(row)

enrich_go <- enrichGO(
  gene= sigGenes,
  OrgDb = org.Hs.eg.db,
  keyType = "ENSEMBL",
  ont = "BP",
  universe = universe,
  qvalueCutoff = 0.05,
  readable=TRUE
)

```


Enrichment analysis 
Try this it works somehow
```{r}
 library(data.table)
 anno <- fread("C:/Users/parth/OneDrive/Desktop/New_start/de_dormancy-MvsD.csv")
 class(anno) <- "data.frame"
 library(clusterProfiler)
 gene <- sample(anno$row, 100)
 TERM2GENE <- anno[, c("baseMean","SYMBOL")]
 TERM2NAME <- anno[, c("baseMean", "SYMBOL")]
 ego <- enricher(gene = gene, TERM2GENE = TERM2GENE,  TERM2NAME = TERM2NAME)
 
 
 head(ego)
  
```



```{r}
head(anno)
```
```{r}
library(DESeq2)
library(pheatmap)
library(org.Mm.eg.db)
library(DOSE)

library(clusterProfiler)
install.packages("ensembl")

library(tidyverse)
library(dplyr)
```
```{r}
all(colnames(DatasheetB)%in%rownames(coldata))
all(colnames(DatasheetB)==rownames(coldata))

```
```{r}
rold<-rlog(dds,blind = TRUE)
plotPCA(rold, intgroup="condition")
```

```{r}
res<-results(dds)
all_genes <- as.character(rownames(res))
signif_res <- res[res$padj < 0.05 & !is.na(res$padj), ]
signif_genes <- as.character(rownames(signif_res))
```

```{r}
library(org.Hs.eg.db)

ego <- enrichGO(gene = signif_genes,
 universe = all_genes,
 keyType = "ENTREZID",
 OrgDb = org.Hs.eg.db,
 ont = "BP",
 pAdjustMethod = "BH",
 qvalueCutoff = 0.05,
 readable = TRUE)
```

```{r}
cluster_summary <- data.frame(ego)
```

```{r}
dotplot(ego, showCategory=50)
```

